fdbserver/QuietDatabase.actor.cpp

```diff
@@ -284,6 +284,52 @@ ACTOR Future<vector<WorkerInterface>> getStorageWorkers(Database cx,
 	return result;
 }
 
+int64_t extractMaxQueueSize(const std::vector<Future<TraceEventFields>>& messages,
+                            const std::vector<StorageServerInterface>& servers) {
+
+	int64_t maxQueueSize = 0;
+	UID maxQueueServer;
+
+	for (int i = 0; i < messages.size(); i++) {
+		try {
+			auto queueSize = getQueueSize(messages[i].get());
+			if (queueSize > maxQueueSize) {
+				maxQueueSize = queueSize;
+				maxQueueServer = servers[i].id();
+			}
+		} catch (Error& e) {
+			TraceEvent("QuietDatabaseFailure")
+			    .detail("Reason", "Failed to extract MaxStorageServerQueue")
+			    .detail("SS", servers[i].id());
+			for (auto& m : messages) {
+				TraceEvent("Messages").detail("Info", m.get().toString());
+			}
+			throw;
+		}
+	}
+
+	TraceEvent("QuietDatabaseGotMaxStorageServerQueueSize")
+	    .detail("Stage", "MaxComputed")
+	    .detail("Max", maxQueueSize)
+	    .detail("MaxQueueServer", format("%016" PRIx64, maxQueueServer.first()));
+	return maxQueueSize;
+}
+
+ACTOR Future<TraceEventFields> getStorageMetricsTimeout(UID storage, WorkerInterface wi) {
+	state Future<TraceEventFields> result =
+	    wi.eventLogRequest.getReply(EventLogRequest(StringRef(storage.toString() + "/StorageMetrics")));
+	state Future<Void> timeout = delay(1.0);
```

Comment:
Should this delay be a knob?

Reply:
I don't think we need a knob for this -- it wasn't a knob before either (before this change we called `timeoutError` and passed `1.0`).